client.go

```diff
@@ -561,6 +564,14 @@ func (client *Client) EventFromCheckIn(checkIn *CheckIn, monitorConfig *MonitorC
 	return event
 }
 
+func (client *Client) SetSDKIdentifier(identifier string) {
+	sdkIdentifier = identifier
```

Comment:
Looking at https://github.com/getsentry/sentry-go/actions/runs/5807554425/job/15742607813, we might need a mutex here 🤔